File: fe2-android/app/src/main/java/com/github/dedis/popstellar/utility/error/keys/KeyException.java
package com.github.dedis.popstellar.utility.error.keys;

import com.github.dedis.popstellar.utility.error.GenericException;

/** This class regroup all the exceptions that can be generated when retrieving a key */
public abstract class KeyException extends GenericException {

  protected KeyException(String msg, Exception cause) {
    super(msg, cause);
  }

  protected KeyException(String msg) {
    super(msg);
  }
}

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/utility/error/keys/KeyException.kt
package com.github.dedis.popstellar.utility.error.keys

import com.github.dedis.popstellar.utility.error.GenericException

/** This class regroup all the exceptions that can be generated when retrieving a key */
abstract class KeyException : GenericException {
  protected constructor(msg: String?, cause: Exception?) : super(msg, cause)

  protected constructor(msg: String?) : super(msg)
}
